### PR TITLE
#19319 improvement

### PR DESCRIPTION
**Pull request by freddyucv**

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/maintenance/MaintenanceResource.java [hunk 1]
```diff
@@ -1,5 +1,7 @@
 package com.dotcms.rest.api.v1.maintenance;
 
+import com.dotcms.concurrent.DotConcurrentFactory;
+import com.dotcms.content.elasticsearch.business.ESReadOnlyMonitor;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/maintenance/MaintenanceResource.java:4]: Codacy found an issue: [Avoid unused imports such as 'com.dotcms.content.elasticsearch.business.ESReadOnlyMonitor'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6380610)

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/maintenance/MaintenanceResource.java [hunk 2]
```diff
@@ -73,7 +75,12 @@ public final Response shutdown(@Context final HttpServletRequest request,
             return Response.status(Status.FORBIDDEN).build();
         }
 
-        Runtime.getRuntime().exit(0);
+        DotConcurrentFactory.getInstance()
+                .getSubmitter()
+                .submit(
+                        () -> Runtime.getRuntime().exit(0)
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/maintenance/MaintenanceResource.java:81]: Codacy found an issue: [System.exit() should not be used in J2EE/JEE apps](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6380610)

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/maintenance/MaintenanceResource.java [hunk 3]
```diff
@@ -73,7 +76,14 @@ public final Response shutdown(@Context final HttpServletRequest request,
             return Response.status(Status.FORBIDDEN).build();
         }
 
-        Runtime.getRuntime().exit(0);
+        DotConcurrentFactory.getInstance()
+                .getSubmitter()
+                .submit(
+                        () -> Runtime.getRuntime().exit(0),
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/maintenance/MaintenanceResource.java:82]: Codacy found an issue: [System.exit() should not be used in J2EE/JEE apps](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6380610)

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/maintenance/MaintenanceResource.java (hunk 1 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/maintenance/MaintenanceResource.java:4]: Codacy found an issue: [Avoid unused imports such as 'com.dotcms.content.elasticsearch.business.ESReadOnlyMonitor'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6380610)

**Review by dsilvam — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/maintenance/MaintenanceResource.java [hunk 4]
```diff
@@ -65,15 +68,22 @@ public final Response shutdown(@Context final HttpServletRequest request,
...
 
         if (!Config.getBooleanProperty("ALLOW_DOTCMS_SHUTDOWN_FROM_CONSOLE", true)) {
             return Response.status(Status.FORBIDDEN).build();
         }
 
-        Runtime.getRuntime().exit(0);
+        DotConcurrentFactory.getInstance()
+                .getSubmitter()
+                .submit(
+                        () -> Runtime.getRuntime().exit(0),
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/maintenance/MaintenanceResource.java:82]: Codacy found an issue: [System.exit() should not be used in J2EE/JEE apps](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6380610)

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/maintenance/MaintenanceResource.java (hunk 1 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/maintenance/MaintenanceResource.java:4]: Codacy found an issue: [Avoid unused imports such as 'com.dotcms.content.elasticsearch.business.ESReadOnlyMonitor'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6380610)

**Pull request merged**